### PR TITLE
Feature/saleor 1564 Add next.js to the Storefront 

### PR DESCRIPTION
**Pull request by piotrgrundas**
Resolves [Switch Storefront to next.js](https://github.com/mirumee/saleor-storefront/pull/962) 
(client routing only)

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Changes are mentioned in the changelog.

### Test Environment Config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by matiasmm** [comment]
@piotrgrundas what's the state of this?  is this going to get merged to master soon?

**Comment by piotrgrundas** [comment]
@matiasmm that's the plan. 
The second PR is waiting to be merged which will finish adding nextjs client only routing to the storefront. Should be merged in the upcoming days.

**Review by jwm0 — CHANGES_REQUESTED** [review]
Please check the LGTM bot issue and check what's causing the sudden waterfall of TS issues (it looks like a strict TS config is targeting files outside the `src/next` directory)

File: .github/workflows/test-env-deploy.yml [hunk 1]
```diff
@@ -60,12 +60,17 @@ jobs:
...
           echo "::set-output name=custom_api_uri::$(echo $pull_request_body | grep -Eo "$prefix$pattern" | sed s/$prefix// | head -n 1)"
 
-      - name: Run build
-        env:
+      - name: Create env file
+        uses: SpicyPizza/create-envfile@v1
+        with:
           # Use custom API_URI or the default one
-          API_URI: ${{ steps.api_uri.outputs.custom_api_uri || 'https://master.staging.saleor.rocks/graphql/' }}
+          envkey_NEXT_PUBLIC_API_URI: ${{ steps.api_uri.outputs.custom_api_uri || 'https://master.staging.saleor.rocks/graphql/' }}
```
Comment (krzysztofwolski) [inline .github/workflows/test-env-deploy.yml:67]:
```suggestion
          envkey_NEXT_PUBLIC_API_URI: ${{ steps.api_uri.outputs.custom_api_uri || 'https://master.staging.saleor.rocks/graphql/' }}
```
```suggestion
          envkey_NEXT_PUBLIC_API_URI: ${{ steps.api_uri.outputs.custom_api_uri || 'https://master.staging.saleor.rocks/graphql/' }}
```
```suggestion
          envkey_NEXT_PUBLIC_API_URI: ${{ steps.api_uri.outputs.custom_api_uri || secrets.API_URI }}
```

**Review by krzysztofwolski — APPROVED** [review]
One issue, other that that lookin good

**Review by jwm0 — APPROVED** [review]

**Pull request merged**